### DISCUSSION TITLE
Keine Anzeige in der Live Ansicht

### DISCUSSION THREAD

**Discussion by iukbox** [hoylabs/OpenDTU-OnBattery#218] [Q&A]
Seit kurzem habe ich keine Anzeige mehr in der Live Ansicht. Es taucht nur kurz das Wort Live Ansicht auf und das war's. 
Bildschirm ist und bleibt weiß.
Nach einem Neustart der DTU ist die Seite kurzfristig zu sehen, später wieder nichts angezeigt.
Auf verschiedenen Geräten probiert, überall gleiches Erscheinungsbild. Cache wurde auch geleert.
Software ist aktuell.
Irgendjemand eine Idee?

**Comment by iukbox** [comment]
Hab Mal in die Konsole geschaut, hilft das?


RX Period End
12:14:15.205 > Success
12:14:15.775 > Websocket: [/livedata][20] disconnect
12:14:16.845 > [PowerLimiterClass::loop] dcVoltage: 32.20 Voltage Start Threshold: 10.00 Voltage Stop Threshold: 1.00 inverter->isProducing(): 1
12:14:16.845 > [PowerLimiterClass::loop] newPowerLimit: 1500
12:14:20.076 > Fetch inverter: 116183307544
12:14:20.090 > TX RealTimeRunData Channel: 61 --> 15 83 30 75 44 80 14 36 36 80 0B 00 64 53 85 7C 00 00 00 00 00 00 00 00 FA F9 45 
12:14:20.121 > Interrupt received
12:14:20.126 > RX Channel: 3 --> 95 83 30 75 44 83 30 75 44 02 00 00 8A 40 00 B5 00 BF 01 45 03 9E 03 AD 0B BB 90 
12:14:20.690 > RX Period End
12:14:20.690 > Last missing
12:14:20.690 > Request retransmit: 3
12:14:20.690 > TX RequestFrame Channel: 75 --> 15 83 30 75 44 80 14 36 36 83 80 
12:14:20.795 > Interrupt received
12:14:20.821 > RX Channel: 61 --> 95 83 30 75 44 83 30 75 44 03 0B EB 00 00 87 D2 00 00 8C 48 00 B7 00 C0 09 88 11 
12:14:20.843 > RX Period End
12:14:20.844 > Last missing
12:14:20.844 > Request retransmit: 4
12:14:20.844 > TX RequestFrame Channel: 3 --> 15 83 30 75 44 80 14 36 36 84 87 
12:14:20.874 > Interrupt received
12:14:20.895 > RX Channel: 40 --> 95 83 30 75 44 83 30 75 44 84 13 88 2C AA 00 E8 01 D5 03 E8 01 4B 00 25 23 DA 4D 
12:14:20.907 > RX Period End
12:14:20.908 > Middle missing
12:14:20.908 > Request retransmit: 1
12:14:20.908 > TX RequestFrame Channel: 23 --> 15 83 30 75 44 80 14 36 36 81 82 
12:14:20.910 > Interrupt received
12:14:20.916 > RX Channel: 61 --> 95 83 30 75 44 83 30 75 44 01 00 01 01 43 03 9C 03 A6 0B 9E 0B C0 00 00 87 13 27 
12:14:20.943 > RX Period End
12:14:20.943 > Success

**Comment by helgeerbe** [comment]
What do you get if you call http://opendtu/api/livedata/status ? Replace opendtu with your device name, or ip address.

**Comment by iukbox** [comment]
Sorry, mußte warten bis der Fehler wieder auftritt.
Solange die Liveansicht verfügbar ist, kommen beim API Aufruf alle Daten.
Wenn die Liveansicht nicht mehr verfügbar ist, dann kommt als Rückantwort **null**

Alle anderen Menüpunkte wie WR, NTP, Netzwerk funktionieren von der Weboberfläche.

**Comment by peff74** [comment]
I also have this problem from time to time. I will observe it and report if it happens.

**Comment by helgeerbe** [comment]
Ich vermute, dass es an https://github.com/tbnobody/OpenDTU/issues/739 liegt.

Der esp kann zeitweise keinen zusammenhängenden Speicherblock für das JSON Dokument bekommen. Das liegt an der Speicherfragmentierung. Einzige Lösung, die mir aktuell einfällt, die Anzahl der zulässigen Inverter von 10 auf 5 reduzieren und die Speicheranfragen halbieren. Das wäre aber unschön.

Oder die original openDTU versucht, nicht mehr den Speicher für das JSOn Dokument auf einmal zu alloziieren, sondern sendet die Daten häppchenweise.  Aber das müsste erst mal jemand implementieren.

**Comment by iukbox** [comment]
Nun ja, die max Bytes zu reservieren ist ja soweit ok, aber wer hat 10 Wechselrichter zu administrieren die dann auch noch in Funkreichweite der DTU liegen?
Ich könnte problemlos mit der Hälfte leben. Wenn dafür die Software ohne Bug läuft.

Btw, kann man eigentlich mehr als einen Victron anschließen? Mein geplantes zukünftiges Setup sieht so aus das ich 2 Victron nutzen möchte um alle 10 Paneele zum Laden zu nutzen.

**Comment by helgeerbe** [comment]
Anscheinend gibt es openDTU user, für die selbst 10 Inverter zu wenig ist. 

Ich bin noch am überleben, ob ich hier vom Original abweichen werden.

Mehrere Victrons sind geplant. Nur mir fehlt aktuell die Zeit.

**Reply by swingstate** [reply]
wenn ich doch nur programmieren könnte. So viele Ideen, das würde fliegen 😸

**Comment by madmartin** [comment]
Mir ist aufgefallen daß das mit der Anzahl der laufenden Browser-Fenster zusammenhängt, die auf den Live-View "gerichtet sind". 
Bei drei Computern mit geöffnetem Browser kommt es gelegentlich zu Problemen, schließe ich einen Browser (oder lege den PC schlafen) läuft es bei den anderen besser.
Vielleicht sollte der Output für den Live-View gecached werden und nicht für jede Session neu zusammengebaut werden?

**Comment by Manos1966** [comment]
Ich habe das gleiche Problem, 
ich habe zwei OpenDTUonBattery, Live Ansicht bleib max 5 Min lang und danach verschwindet :-(

Auf beide dind die gleiche zwei Wechselrichter angeschlossen, auf die Daten einer SHELLY 3EM wird zugegriffen per JSON, und die Daten werden per MQTT an einer Datebank geschrieben.

Ich sollte vielleicht erwaehnen, ich es handelt sich um eine Test Installation die ich in einem Photovoltaik "Lagerhaus" teste (wenn ich zufrieden mit dem Ergebniss bin, kann ich das komplete System kaufen).
Insgesamt sind eine OpenDTU, zwei OpenDTUonBattery (meine) und eine Hoymiles DTU aktiv, und schaetschungsweise 6-7 Hoymiles Wechselrichter (bin nicht sicher, ich muss im Laden nachfragen).
So oder so, nur zwei Wechselrichter sind an meine zwei OpenDTUonBattery angeschlossen. 
Aber vielleicht, die Tatsache dass so viele Hoymiles senden, verursacht Probleme?
Zur Erlaeuterung, ich hatte nur eine OpenDTUonBattery am Anfang. Ich habe eine zweite angeschlossen nur als Test, als ich dachte dass etwas mit meiner HW nicht stimmt. Als die zweite sofort das gleiche Verhalten gezeigt hat (binnen 5 Min Live Ansicht verschwindet), wusste ich dass etwas nicht stimmt.

**Comment by helgeerbe** [comment]
Habe jetzt die maximale Anzahl der Inverter auf 5 gesetzt. Hoffe, damit ist das Problem behoben. Release 2023.05.23.post3.

**Reply by Manos1966** [reply]
Hallo @Helgeerbe,
vielen Dank fuer die neue Version (Release 2023.05.23.post3)
Ich will ein Update geben: Ich glaube NICHT dass meine Probleme etwas mit der max. erlaubte Anzahl Wechselrichter zu tun hatten.
Deine SW war NICHT der Grund. 👍 
Ich habe das problematische Geraet getestet und festgestellt, dass das nRF24+ Kommukationsmodul  defekt war!!!
Ich habe das nRF24+ Modul ersetzt, danach lief alles wieder problemlos :-) 
Warum es ist wichtig? 
Weil, beim Display "**Power Safe aktivieren:**" eingeschaltet war! 
Mit nRF24 defekt = kein Kontakt zu den Wechselrichtern = keine Stromerzeugung = **das Display schaltet sich automatisch ab!** :-D

**Comment by spcqike** [comment]
rein zur Info, eventuell ist das Problem auch bereits in der original OpenDTU und tauchte nur hier, aufgrund von heap/RAM/was auch immer, früher auf?
https://github.com/tbnobody/OpenDTU/issues/1000

zumindest hab ich auch gelegentlich das Problem, dass der LiveView nicht mehr geht. ich gucke eigentlich nur mit einem Gerät (Firefox am PC oder Safari am iPhone) gleichzeitig drauf. 

Gestern war es sogar so verrückt, dass ich nicht mal mehr den Neustart durchführen konnte. wie auch hier bei anderen Usern
https://github.com/tbnobody/OpenDTU/issues/1000#issuecomment-1576358786

nach ner Stunde oder so hatte sich die openDTU dann aber von selbst neugestartet 😄

**Comment by helgeerbe** [comment]
> nach ner Stunde oder so hatte sich die openDTU dann aber von selbst neugestartet

die originale openDTU oder OpenDTU-onbattery?  Bei der originalen kenne ich das Problem. Es liegt meiner Meinung nach an der Speicherfragmentierung. Und es gibt einige dynamische new() calls, die keine Exception Behandlung haben. D. h. die DTU startet neu, wenn kein genügend großer, zusammenhängender Speicherblock geholt werden kann. Durch die Reduktion bei OpenDTU-onBattery auf max 5 Inverter, ist mir dieses Fehlerbild allerdings nicht mehr untergekommen.

**Reply by spcqike** [reply]
ne, schon die OpenDTU-OnBattery. 

halt meine geforkte Version, die ja noch den PR hat. Das war zu dem Zeitpunkt auch die einzige Änderung zu der hier verfügbaren Version.

mir ist der Fehler auch bisher nur am Handy passiert, also mit Safario am iPhone. Wenn er da auftritt, dann ist er aber auf allen Geräten.

Gestern war auch das erste und einzige mal, wo ich keinen Reset mehr im WebUI machen konnte.

**Comment by Manos1966** [comment]
Jetzt erlebe ich genau das gleiche Problem.... 

> keine Anzeige mehr in der Live Ansicht. 
Es taucht nur kurz das Wort Live Ansicht auf und das war's.
Bildschirm ist und bleibt weiß.
Nach einem Neustart der DTU ist die Seite kurzfristig zu sehen, später wieder nichts angezeigt.

**Reply by helgeerbe** [reply]
welche version hast du laufen?

**Reply by Manos1966** [reply]
die aktuellste (vom 21.06.2023)

Das Problem hatte bereits mit der (opendtu-onbattery-generic V23.06.14.post2) Version angefangen 
Damals war Live View weg, aber das System funktionierte im Hintergrund.

Jetzt (opendtu-onbattery-generic V2023.06.21):
Live View verschwindet nach ein paar Minuten, System funktioniert im Hintergrund
10-15min spaeter haengt das ganze System: 
MQTT Daten werden nicht mehr gesendet
IP der OpenDTU kann ich nicht mehr pingen
Ich muss den Stecker entfernen, damit das System neustartet.
Ich werde jetzt eine aeltere Firmware installieren...

Meine Konfiguration:
Zwei Hoymiles ("Batterie" & "TestAnlage")
Shelly 3em
Victron
MQTT

anbei die config Datei (ich habe passwoerte, SerNr usw entfern) falls es hilft:
`
{"cfg":
"hostname":"OpenDTU-%06X"},"ntp":{"server":"pool.ntp.org","timezone":"CET-1CEST,M3.5.0,M10.5.0/3","timezone_descr":"Europe/Berlin","latitude":50.6393013,"longitude":7.346650124,"sunsettype":1},"mqtt":{"enabled":true,"hostname":"192.168.178.52","port":1884,,"topic":"solar/","retain":true,"publish_interval":5,"lwt":{"topic":"dtu/status","value_online":"online","value_offline":"offline"},"tls":{"enabled":false,"root_ca_cert":"
,"certlogin":false,"client_cert":"","client_key":""},"hass":{"enabled":false,"retain":true,"topic":"homeassistant/","individual_panels":false,"expire":true}},"dtu":{"serial":,"poll_interval":5,"nrf_pa_level":2,"cmt_pa_level":0,"cmt_frequency":865000},"security":{"password":"openDTU42","allow_readonly":true},"device":{"pinmapping":"","display":{"powersafe":true,"screensaver":true,"rotation":2,"contrast":60,"language":0}},"inverters":[{"serial":,"name":"Batterie","order":0,"poll_enable":true,"poll_enable_night":true,"command_enable":true,"command_enable_night":true,"channel":[{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0}]},{"serial":,"name":"Testanlage Garten","order":0,"poll_enable":true,"poll_enable_night":true,"command_enable":true,"command_enable_night":true,"channel":[{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0}]},{"serial":0,"name":"","order":0,"poll_enable":true,"poll_enable_night":true,"command_enable":true,"command_enable_night":true,"channel":[{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0}]},{"serial":0,"name":"","order":0,"poll_enable":true,"poll_enable_night":true,"command_enable":true,"command_enable_night":true,"channel":[{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0}]},{"serial":0,"name":"","order":0,"poll_enable":true,"poll_enable_night":true,"command_enable":true,"command_enable_night":true,"channel":[{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0},{"name":"","max_power":0,"yield_total_offset":0}]}],"vedirect":{"enabled":true,"updates_only":true,"poll_interval":5},"powermeter":{"enabled":true,"interval":10,"source":3,"mqtt_topic_powermeter_1":"","mqtt_topic_powermeter_2":"","mqtt_topic_powermeter_3":"","sdmbaudrate":9600,"sdmaddress":1,"http_individual_requests":false,"http_phases":[{"enabled":true,"url":"http://192.168.178.96/status?emeters","header_key":"","header_value":"","timeout":1000,"json_path":"total_power"},{"enabled":false,"url":"","header_key":"","header_value":"","timeout":1000,"json_path":""},{"enabled":false,"url":"","header_key":"","header_value":"","timeout":1000,"json_path":""}]},"powerlimiter":{"enabled":true,"solar_passtrough_enabled":true,"battery_drain_strategy":0,"interval":10,"is_inverter_behind_powermeter":true,"inverter_id":0,"inverter_channel_id":0,"target_power_consumption":-30,"target_power_consumption_hysteresis":10,"lower_power_limit":15,"upper_power_limit":1500,"battery_soc_start_threshold":80,"battery_soc_stop_threshold":20,"voltage_start_threshold":53,"voltage_stop_threshold":51.59999847,"voltage_load_correction_factor":0.0013,"inverter_restart_hour":0,"full_solar_passthrough_soc":100,"full_solar_passthrough_start_voltage":54.5,"full_solar_passthrough_stop_voltage":53},"battery":{"enabled":false},"huawei":{"enabled":false}}
`

**Comment by helgeerbe** [comment]
Klingt irgendwie nach Speicherfehler. Die größten Speicherblöcke die allokiert werden, sind die JSON Blöcke für die LiveView. Wenn die nicht allokiert werden können, wird die LiveView nicht mehr angezeigt. 

Wie sieht in der Info->System der Heap aus? Wird der freie Bereich mit der Zeit immer kleiner?

**Reply by Manos1966** [reply]
> Wie sieht in der Info->System der Heap aus? Wird der freie Bereich mit der Zeit immer kleiner?

Ich bin 10min da geblieben und genau das gemonitored, sah aber keine Aenderung. 
Danach bin ich weg gefahren und bis ich zu Hause war (15min spaeter) um es per Remote zu monitoren, war es zu spaet. OpenDTU nicht mehr erreichbar...
Das Geraet hat ein WerksReset gemacht :-o
Ich musste alles neu eigeben.

Am Anfang habe ich gedacht, dass es vielleicht an meiner HW liegt.
Ich benutze (noch) die erste HW, nur ESP32 und NRF24, ohne 100nf Capacitor und ohne Level Shifter 3,3V-5V

Ich habe jetzt die aeltere Firmware opendtu-onbattery-generic V23.06.12 
- Live View ist NICHT sichtbar
- Der Rest aber funktioniert.

anbei die Daten (ca. 45Min nach Start), ich sehe keine Aenderung.
![Memory Information 01 45Min](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/7050baba-0b24-4944-82d5-8a170d800e2a)
![MQTT Information 01 45Min](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/5f20f7ff-290f-4b76-a4e2-667ef9ee1723)

**Reply by Manos1966** [reply]
1Stunde 15min 
![Memory Information 02 1Std15min](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/06c0c7df-87a6-4efd-aa47-df0acad825cb)

**Reply by Manos1966** [reply]
Ich werde es monitoren...
Es ist jetzt die Firmware  opendtu-onbattery-generic V23.06.12 installiert, weil ich fuer ein paar Tage reisen werde und kann nicht vor Ort sein.
![Memory Information 02 1Std15min Graph](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/e356c541-0f7a-4a99-abbe-151b4aadbc86)

**Reply by helgeerbe** [reply]
Das ist gut. Eine gewisse Varianz ist zu erwarten. Aber der Speicherverbrauch sollte nicht kontinuierlich zunehmen.

**Reply by Manos1966** [reply]
Es hat sich beruehigt (jetzt ist der VICTRON aus, sowie der zweite Wechelrichter).
Auf die Live Ansicht der DTU konnte ich trotzdem nicht (blieb leer).
Firmware opendtu-onbattery-generic V23.06.12 

Und jetzt gerade habe ich die OpenDTU wieder "zerschossen" :-(
Ich habe mehrmals sehr schnell auf die "Refresh" Taste des Browsers gedruckt, als ich versuchte die Live Ansicht zu testen. Danach konnte ich nicht mal die Einstellungen sehen und jetzt ist die DTU nicht mehr im WiFI Netz.
Bis dahin war alles OK

![Memory Information 03 9Std](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/2a848c5c-1914-4b95-ab97-6a023b9c8d80)
![WhatsApp Image 2023-06-22 at 21 56 24](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/c2ce0dab-be7a-4a50-81a6-bdcfdb06ddd9)

**Reply by Manos1966** [reply]
Warum sollte ich eine Fehlermeldung "Failed to parse Data" beim Versuch die OpenDTU auszuschalten, bekommen?
Live View ist leer, der Rest (Info, Einstellungen) funktioniert.
 
![failed to parse data](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/38673ea8-d93a-4fa7-8545-4e3de3710af9)
![failed to parse data 2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/00462142-6e88-4525-8302-8537a5766dc2)
![failed to parse data 3](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ff0e8463-f85f-49ad-9b39-f8d4ef526573)

**Reply by iukbox** [reply]
Das hatte ich bei mir auch, kurz bevor sie sich komplett aufgehängt hat.
Ich bin jetzt auch wieder auf eine alte funktionierende Version zurück. Ich teste gerade durch ab welcher Version sich der Fehler eingeschlichen hat.
Bin jetzt bei Version [40a6519]

**Reply by Manos1966** [reply]
Live view is back again. Reboot accepted (Reboot trigerred)
I noticed that it did a reset (Uptime is only 55minutes)!
There seems to be a problem with the connection Data Age 1947seconds
There is a strange recurring increase in Memory Heap but even stranger, the Max Memory number is variating as well (it should not be).
There is also another strange information, MQTT says Limit 24W but the inverter seems to be producing 190W
On Wednesday when I return to Germany, I will change the HW and install Helger's pcb with the ELKO and KERKO, maybe it is a HW error.
![failed to parse data b1](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/4d8b22d5-8ed9-4eea-8104-2977d4a2a748)
![failed to parse data b2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/f40c69f6-7502-40f3-b002-427c804336eb)
![failed to parse data b3](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/539c29b7-4ede-42d7-acd9-74fa7b27fb5e)
![failed to parse data b4](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ab0a7a6b-23b4-4d78-9698-951d344a74ad)

**Comment by Manos1966** [comment]
This is interesting,
The total Heap size is variating (now 259kB). If you look at some pictures above, it was 265kB.
I am still using Firmware Version / Git Hash  8dac88e, when I am back in Germany I will install the new test hardware and the newest firmware (and hopefully the Pylontech battery will arrive as well).
![failed to parse data c1](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/2c2603f5-bf80-4e6a-8efd-fc720e0142ac)

**Comment by Manos1966** [comment]
The new Firmware (26.06.2023) did not help.
I rolled back to Firmwareversion / git Hash  fd58ad2, this one seems to be working without problems but I did the change in the evening when VICTRON and the second HOYMILES were off so, I have to let the system run for a few days with sunshine before I can make any comments.

I need to ask though: 
- Is there a way to monitor the CPU usage?

- Heap! Why would the Heap Max size be different?
Firmwareversion / git Hash  fd58ad2 Heap Max **270kB**
![Memory Information git Hash  fd58ad2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/f303da5d-098d-457b-9655-afcd6eb0990e)

Firmwareversion  V23.06.12 Heap Max was only **265kB**
![Memory Information 02 1Std15min](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/a063eeab-aa60-4309-910d-e0811b6cf624)

**Comment by Manos1966** [comment]
@helgeerbe  Update,
Firmwareversion / git Hash https://github.com/helgeerbe/OpenDTU-OnBattery/commit/fd58ad2003a9ad9c55dfab962aa09575badced0c

Heap memory remains stable (Frei: 130 kB  **Benutzt: 140 kB**  Groesse: 270 kB), with all systems running.

**Now that I look at it, there is a significant increase in the usage of Heap Memory in the newer FW versions (it goes from 140kB up to 203kB).**

If it helps, this is my setup(s):
ESP32 Dev Kit C V4 38pin (mostly with CP2102)
2,4G 1100 mt NRF24L + PA + LNA mit Antenne
(noch kein ELKO, KERKO soldered)

Victron (VE.Direct)
Hoymiles HM-1500
Hoymiles HM-800
Shelly 3em (over HTTP+JSON)
Pylontech with CANbus (soon)
Battery 2 x 25V = 51V with BMS
MQTT connection
My Homematic also does JSON requests every minute:
192.168.xxx.xxx/api/system/status
192.168.xxx.xxx/api/powerlimiter/status

**Comment by helgeerbe** [comment]
That the heap size varies slightly, seem to be normal, when you change the firmware.  But it should not change during runtime.

**Comment by helgeerbe** [comment]
You mentioned  V23.06.12. Versions with a v prefix indicate the original OpenDTU. OpenDTU-onBattery is using the complete date schema like "2023.06.26". And yes, there should be a difference in heap usage between openDTU and openDTU-onBattery. I reduced the max. number of inverters from 10 to 5 to have more heap memory available for onBattery extensions.

**Reply by Manos1966** [reply]
sorry for the misunderstanding Helge, I change the naming of the files in order to better sort them.
All are OpenDTU-onBattery versions (no OpenDTU versions).

There is a big increase in the usage of Heap Memory in the newer OpenDTU-OnBattery FW versions (it goes from 140kB up to 203kB). - With the same configuration as mentioned in my previous post.
See the Heap memory usage from my post from last week:
![Memory Information V230612](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/c2959c5d-1645-4d68-9f3e-4e54c2109863)

**Comment by helgeerbe** [comment]
My usage is about 170 kb.  Our config is quite similar, but I don't have Pylontech with CANbus,

What happens if you switch off all onBattery extensions and do a reboot? Is the heap usage still so high?

If not, switch step by step the extensions on and do a reboot on each step. Is there a significant jump in the heap usage?

**Reply by Manos1966** [reply]
I will start testing and report...

I was going through the PIN settings and noticed I have a difference from the VICTRON standard pin-out.
I am using pins 22 and 1 because this is in one of the first schematics I found in this forum (see picture below).
Can it be that the Pin 1 (TX0) is causing a problem?
![00 My VICTRON PIN  Wiring to match Stecker](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/bf50be07-edbd-4862-b711-63726557d588)

**Comment by helgeerbe** [comment]
The answer is I don't know.  But in general it is not a good idea to use TX0, if there are free pins available. TX0 might be used as output pin (but not as input pin). So here you are save. But during boot, there is debug output on the pin. I don't know how the victron react on this.

**Reply by Manos1966** [reply]
Much appreciated Helge.
I will change it.
It would be good to mention something or update the graphics as well:
 
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/0ef675b7-3cc4-4c40-8c9c-5d80941e4aaa)

**Comment by Manos1966** [comment]
New Firmware from 26.06.2023  Firmwareversion / git Hash  2223afa
I am happy I posted pictures of all my trouble with this firmware because I might have thought I am crazy...

Installed everything from the beginning, updated the pin-out before connecting the OpenDTU to Victron, MQTT etc etc.
The result is, Heap memory is stable (after 3hrs of running) at  130-140kB - Big difference from the 200kB I had before!
Thanks @helgeerbe for the explanations regarding GPIO 1/TX0, maybe this solved the problem.

Anyway, "better be lucky, than good, any day!" I can not explain what I did correctly, anyhow it is working now.
The Live View is working without problems.
![Memory Information git Hash  2223afa solved](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/5fe076cd-8573-48cb-86d6-a90e468e64a0)
